Source/Libraries/GSF.Web/Security/AuthenticationHandler.cs

```diff
@@ -149,8 +149,12 @@ protected override Task<AuthenticationTicket> AuthenticateCoreAsync()
             {
                 // Pick the appropriate authentication logic based
                 // on the authorization type in the HTTP headers
+                // or in the URI Parameters if it is using OIDC.
                 if (authorization?.Scheme == "Basic")
                     securityPrincipal = AuthenticateBasic(authorization.Parameter);
+                // If the resources contains a code make an Attempt to Authorize via OIDC Auth server
+                else if (Request.QueryString.HasValue && Request.QueryString.Value.Contains("code="))
```

Comment:
This has a high probability of becoming a false positive (same below). Like better to parse QueryString parameters into a HashSet or Dictionary (function likely already exists) and do a `.Contains("code")` instead. Even then, the odds of just having a "code" parameter are high. Is there any other distinguishing characteristic that indicates OIDC, like a header value? Or is the code formatted in such a way that it has a detectible pattern?

Reply:
I will change the logic to use a Dictionary or hashset. 
The reason I am not too concerned about false positives here is :
if the requested resource is available anonymous it should never get to this point. 
If the user is logged in trying to access a resource it should not get to this point either. 

The general flow for OIDC is User requests resource -> Application responds with Redirect to auth server and generates a key -> Auth Server lets user log in and redirects back to http GET endpoint with `code` parameter (which is this point) 
We could change the logic by requiring that endpoint to be set in the config file instead of looking for the `code` parameter.